xc/domain.ml

```diff
@@ -683,6 +740,20 @@ let build_pre ~xc ~xs ~vcpus ~memory domid =
       Xenctrl.shadow_allocation_set xc domid shadow_mib
     );
 
+  if !Xenopsd.numa_placement then
+    log_reraise (Printf.sprintf "NUMA placement") (fun () ->
+        if has_hard_affinity then begin
+          D.debug "VM has hard affinity set, skipping NUMA optimization"
```

Comment:
no need for `begin ... end`

Reply:
I prefer to use `begin ... end` on both arms of an `if` if any of them requires it, in case more statements need to be added in the future.